processes/ingest/clacso.py

```diff
@@ -1,45 +1,106 @@
 import os
-from services import CLACSOService
+import json
+import dataclasses
+from services import DSpaceService
 
 from ..core import CoreProcess
 from logger import create_log
-from managers import S3Manager
+from managers import S3Manager, WebpubManifest
+from mappings.clacso import CLACSOMapping
+from model import Part, FileFlags
+
 
 logger = create_log(__name__)
 
 class CLACSOProcess(CoreProcess):
+    CLACSO_BASE_URL = 'https://biblioteca-repositorio.clacso.edu.ar/oai/request?'
+
     def __init__(self, *args):
         super(CLACSOProcess, self).__init__(*args[:4])
 
+        self.s3_manager = S3Manager()
+        self.s3_manager.createS3Client()
+        self.s3_bucket = os.environ['FILE_BUCKET']
+
+        self.file_queue = os.environ['FILE_QUEUE']
+        self.file_route = os.environ['FILE_ROUTING_KEY']
+
         self.offset = int(args[5]) if args[5] else 0
-        self.limit = (int(args[4]) + self.offset) if args[4] else 1000
+        self.limit = (int(args[4]) + self.offset) if args[4] else 10000
 
-        self.clacso_service = CLACSOService()
+        self.dspace_service = DSpaceService(base_url=self.CLACSO_BASE_URL, source_mapping=CLACSOMapping)
         
     def runProcess(self):
         try:
             self.generateEngine()
             self.createSession()
 
+            records = []
+
             if self.process == 'daily':
-                records = self.clacso_service.get_records()
+                records = self.dspace_service.get_records(offset=self.offset, limit=self.limit)
             elif self.process == 'complete':
-                records = self.clacso_service.get_records(full_import=True, offset=self.offset, limit=self.limit)
+                records = self.dspace_service.get_records(full_import=True, offset=self.offset, limit=self.limit)
             elif self.process == 'custom':
-                records = self.clacso_service.get_records(start_timestamp=self.ingestPeriod, offset=self.offset, limit=self.limit)
+                records = self.dspace_service.get_records(start_timestamp=self.ingestPeriod, offset=self.offset, limit=self.limit)
             else: 
                 logger.warning(f'Unknown CLACSO ingestion process type {self.process}')
                 return
-            for record in records:
-                self.addDCDWToUpdateList(record)
             
+            if records:
+                for record in records:
+                    self.store_pdf_manifest(record)
+                    self.addDCDWToUpdateList(record)
+
             self.saveRecords()
             self.commitChanges()
 
-            logger.info(f'Ingested {len(records)} CLACSO records')
+            logger.info(f'Ingested {len(self.records) if records else 0} CLACSO records')
```

Comment:
Recommend that y'all move over to using the RecordBuffer and not inherit from the CoreProcess class.

Reply:
Ok we'll try implementing the RecordBuffer.